domain/relation/service/relation.go

```diff
@@ -414,7 +429,14 @@ func (s *Service) GetRelationUnitByID(
 	relationID int,
 	unitName unit.Name,
 ) (corerelation.UnitUUID, error) {
-	return "", coreerrors.NotImplemented
+	if err := unitName.Validate(); err != nil {
+		return "", errors.Capture(err)
+	}
+	uuid, err := s.st.GetRelationUUIDByID(ctx, relationID)
```

Comment:
Would be nice to do this all in one tx, but to be honest I'm not sure its worth taking the time to optimise now